crates/net/network/src/metrics.rs

```diff
@@ -244,3 +246,45 @@ pub struct EthRequestHandlerMetrics {
     /// Number of received bodies requests
     pub(crate) received_bodies_requests: Counter,
 }
+
+/// Eth67 announcement metrics, track entries by TxType
+#[derive(Metrics)]
+#[metrics(scope = "network.transaction_fetcher")]
+pub struct AnnouncedTxSizeMetrics {
+    /// Histogram for tracking legacy transaction sizes
+    pub(crate) legacy: Histogram,
+
+    /// Histogram for tracking EIP-2930 transaction sizes
+    pub(crate) eip2930: Histogram,
+
+    /// Histogram for tracking EIP-1559 transaction sizes
+    pub(crate) eip1559: Histogram,
+
+    /// Histogram for tracking EIP-4844 transaction sizes
+    pub(crate) eip4844: Histogram,
+}
+
+impl AnnouncedTxSizeMetrics {
+    /// Update metrics during announcement validation, by examining each announcement entry based on
+    /// TxType
+    pub(crate) fn update_on_valid_announcement(&self, announcement: &ValidAnnouncementData) {
```

Comment:
can we get metrics in the name here too for readability
```suggestion
    pub(crate) fn update_eth68_announcement_metrics(&self, announcement: &ValidAnnouncementData) {
```